### DISCUSSION TITLE
A plugin manager called `wplug`

### DISCUSSION THREAD

**Discussion by kostya-zero** [wezterm/wezterm#4128] [Show and tell]
I finally released my plugin manager for WezTerm - `wplug`. `wplug` at this moment can install and update plugins. I plan to extend its functionality for user and developers. I have also made configuration similar to Neovim's package managers. Any suggestions are welcome.

An example of its configuration:
```lua
local wplug = require("wplug")
wplug.setup({
    spec = {
        {
            "kostya-zero/wplug",
            provider = "https://gitlab.com",
            branch = "main",
            name = "wplug"
        }
    }
})
```

`wplug` installs itself and all plugins in `$HOME/.wezterm` to update with other plugins.

Repository: https://github.com/kostya-zero/wplug

**Comment by wez** [comment]
Thanks for sharing. Did you see https://github.com/wez/wezterm/commit/e4ae8a844d8feaa43e1de34c5cc8b4f07ce525dd which is wezterm's built-in plugin management?
It's not documented in the docs at this time

**Reply by kostya-zero** [reply]
Yeah, I have seen that. I just want to make my own plugin manager for WezTerm that works the same way as plugin managers for Neovim. I was also looking for a solution to integrate it with the built-in plugin manager, but at the moment I'm focused on implementing new features.

**Comment by 0xJWLabs** [comment]
@kostya-zero I know it’s been almost a year, do you still have it? since repo doesn’t exist anymore.

**Reply by kostya-zero** [reply]
I stopped using wezterm a year ago, so I don't have source code anymore.

**Reply by 0xJWLabs** [reply]
ahh, alright, thanks!
